add Qt6::GuiPrivate component for rendergraph QShader support

mxmilkiib opened the pull request:
fixes #15485

## issue
rendergraph opengl backend fails to compile on fresh Qt6 builds with:
```
fatal error: rhi/qshader.h: No such file or directory
```

## root cause
the rendergraph opengl backend requires Qt6::GuiPrivate for QShader support (rhi/qshader.h), but it was never added to the main cmake component list when rendergraph was introduced in b9080f1083.

the header exists in Qt6 installations at:
```
/usr/include/qt6/QtGui/<version>/QtGui/rhi/qshader.h
```

but cmake can only find it when Qt6::GuiPrivate is explicitly listed in find_package() components, which adds the private header paths.

## fix
added `GuiPrivate` to `QT_EXTRA_COMPONENTS` list for Qt6 builds.

## testing
verified that the build now succeeds on Arch Linux with Qt 6.10.

daschuer reviewed (changes requested):
Thank you for the PR. Can you rebase it to the 2.5 branch?

JoergAtGithub commented on CMakeLists.txt:
```diff
@@ -3515,6 +3515,7 @@ set(
 set(QT_EXTRA_COMPONENTS "")
 if(QT6)
   find_package(QT 6.2 NAMES Qt6 COMPONENTS Core REQUIRED)
+  list(APPEND QT_EXTRA_COMPONENTS "GuiPrivate")
```
Is 6.9 also affected or only 6.10?

daschuer replied:
GuiPrivate exists from 6.9. There is however no build failure with 6.9. Maybe this component is dragged in implicit. Making it explicit for 6.9 can't hurt.

Commit:
add Qt6::GuiPrivate component for rendergraph QShader support (Qt >= 6.10)

fixes #15485

the rendergraph opengl backend requires Qt6::GuiPrivate for QShader
support (rhi/qshader.h), but it was never added to the main cmake
component list when rendergraph was introduced in b9080f1.

this is a regression from Qt 6.10 onwards where the header exists at:
/usr/include/qt6/QtGui/<version>/QtGui/rhi/qshader.h

cmake can only find it when Qt6::GuiPrivate is explicitly listed in
find_package() components, which adds the private header paths.

conditionally adds GuiPrivate to QT_EXTRA_COMPONENTS for Qt >= 6.10.

daschuer commented:
It is probably the easiest way to close this PR and create one from a new branch on top of 2.5. 
If you like to fiddle with git, you can try. 
Accept the suggestion, an then:
```
git pull 
git rebase --onto=upstream/2.5 HEAD~2
git push -f 
```

daschuer reviewed (approved):
LGTM, thank you.
